src/components/TodoList/TodoList.tsx

```diff
@@ -1,100 +1,108 @@
-        <td className="is-vcentered is-expanded">
-          <p className="has-text-danger">delectus aut autem</p>
-        </td>
-        <td className="has-text-right is-vcentered">
-          <button data-cy="selectButton" className="button" type="button">
-            <span className="icon">
-              <i className="far fa-eye" />
-            </span>
-          </button>
-        </td>
-      </tr>
-      <tr data-cy="todo" className="has-background-info-light">
-        <td className="is-vcentered">2</td>
-        <td className="is-vcentered" />
-        <td className="is-vcentered is-expanded">
-          <p className="has-text-danger">quis ut nam facilis et officia qui</p>
-        </td>
-        <td className="has-text-right is-vcentered">
-          <button data-cy="selectButton" className="button" type="button">
-            <span className="icon">
-              <i className="far fa-eye-slash" />
-            </span>
-          </button>
-        </td>
-      </tr>
+export const TodoList: React.FC<Props> = ({ setModel }) => {
+  const {
+    todos,
+    filterOption,
+    query,
+    selectedIdTodo,
+    setSelectedIdTodo,
+  } = useContext(TodoContext);
 
-      <tr data-cy="todo" className="">
-        <td className="is-vcentered">1</td>
-        <td className="is-vcentered" />
-        <td className="is-vcentered is-expanded">
-          <p className="has-text-danger">delectus aut autem</p>
-        </td>
-        <td className="has-text-right is-vcentered">
-          <button data-cy="selectButton" className="button" type="button">
-            <span className="icon">
-              <i className="far fa-eye" />
-            </span>
-          </button>
-        </td>
-      </tr>
+  const getVisibleTodos = () => {
+    switch (filterOption) {
+      case FilterOption.Active:
+        return todos.filter(todo => !todo.completed
+          && todo.title.toLocaleLowerCase()
+            .includes(query.toLocaleLowerCase()));
 
-      <tr data-cy="todo" className="">
-        <td className="is-vcentered">6</td>
-        <td className="is-vcentered" />
-        <td className="is-vcentered is-expanded">
-          <p className="has-text-danger">
-            qui ullam ratione quibusdam voluptatem quia omnis
-          </p>
-        </td>
-        <td className="has-text-right is-vcentered">
-          <button data-cy="selectButton" className="button" type="button">
-            <span className="icon">
-              <i className="far fa-eye" />
-            </span>
-          </button>
-        </td>
-      </tr>
+      case FilterOption.Completed:
+        return todos.filter(todo => todo.completed
+          && todo.title.toLocaleLowerCase()
+            .includes(query.toLocaleLowerCase()));
+
+      case FilterOption.All:
+        return todos.filter(todo => todo
+          .title.toLocaleLowerCase().includes(query.toLocaleLowerCase()));
+
+      default:
+        return todos;
+    }
+  };
```

Comment:
looks like we can move this function out of the component and avoid declaring it on each rendering